src/common/transformations/tests/common_optimizations/matmul_multiply_fusion.cpp

```diff
@@ -77,6 +79,44 @@ TEST_F(TransformationTestsF, MatMulMultiplyFusionConstantTransposedWeightsNonSca
 }
 
 TEST_F(TransformationTestsF, MatMulMultiplyFusionNonConstantTransposedWeightsNonScalarConstant) {
+    {
+        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{2, 3});
+        auto weights = opset8::Constant::create(element::f32, Shape{2, 3}, {1});
+        auto matmul = std::make_shared<opset8::MatMul>(data, weights, false, true);
+        auto mul_const = opset8::Constant::create(element::f32, Shape{1, 2}, {1, 1});
+        auto mul = std::make_shared<opset8::Multiply>(matmul, mul_const);
+        model = std::make_shared<Model>(NodeVector{mul}, ParameterVector{data});
+
+        manager.register_pass<ov::pass::MatMulMultiplyFusion>();
+    }
+
+    {
+        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{2, 3});
+        auto weights = opset8::Constant::create(element::f32, Shape{2, 3}, {1});
+        auto matmul = std::make_shared<opset8::MatMul>(data, weights, false, true);
+        model_ref = std::make_shared<Model>(NodeVector{matmul}, ParameterVector{data});
+    }
+    comparator.enable(FunctionsComparator::CmpValues::CONST_VALUES);
+}
+
+TEST_F(TransformationTestsF, MatMulMultiplyFusionConstantWeightsMarkedToKeepSrcPrecision) {
+    {
+        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{1, 2, 4, 3});
+        auto weights = opset8::Constant::create(element::i8, Shape{3, 2}, {1, 2, 3, 4, 5, 6});
+        auto mul_dequantized = std::make_shared<opset8::Multiply>(weights, weights);
+        mark_as_dequantization_node(mul_dequantized);
+        auto convert = std::make_shared<ov::op::v0::Convert>(mul_dequantized, ov::element::f32);
+        auto matmul = std::make_shared<opset8::MatMul>(data, convert);
+        auto mul_const = opset8::Constant::create(element::f32, Shape{1, 1, 1, 2}, {2, 3});
+        auto mul = std::make_shared<opset8::Multiply>(matmul, mul_const);
+        model = std::make_shared<Model>(NodeVector{mul}, ParameterVector{data});
+
+        manager.register_pass<ov::pass::MatMulMultiplyFusion>();
+    }
```

Comment:
With this suggestion we will get ideal dequantization graph as in real scenarios
```suggestion
    {
        auto data = std::make_shared<opset8::Parameter>(element::f32, Shape{1, 2, 4, 3});
        auto weights = opset8::Constant::create(element::i8, Shape{3, 2}, {1, 2, 3, 4, 5, 6});
        auto dequantization_convert = std::make_shared<ov::op::v0::Convert>(weights, ov::element::f32);
        auto dequantization_scale = opset8::Constant::create(element::f32, Shape{}, {2});
        auto dequantization_multiply = std::make_shared<opset8::Multiply>(dequantization_convert, dequantization_scale );
        mark_as_dequantization_node(dequantization_multiply);
        auto matmul = std::make_shared<opset8::MatMul>(data, dequantization_multiply);
        auto mul_const = opset8::Constant::create(element::f32, Shape{1, 1, 1, 2}, {2, 3});
        auto mul = std::make_shared<opset8::Multiply>(matmul, mul_const);
        model = std::make_shared<Model>(NodeVector{mul}, ParameterVector{data});

        manager.register_pass<ov::pass::MatMulMultiplyFusion>();
    }
```